pkg/appgw/internaltypes.go

```diff
@@ -88,40 +90,40 @@ func getResourceKey(namespace, name string) string {
 }
 
 func generateHTTPSettingsName(serviceName string, servicePort string, backendPortNo int32, ingress string) string {
-	return governor(fmt.Sprintf("%s-%v-%v-bp-%v-%s", agPrefix, serviceName, servicePort, backendPortNo, ingress))
+	return governor(fmt.Sprintf("%s%v-%v-bp-%v-%s", agPrefix, serviceName, servicePort, backendPortNo, ingress))
```

Comment:
keep the dash ?

Reply:
I deliberately removed the dash here.
When agPrefix is blank, which is our new default, if we keep the dash, all property names will start with a dash.
So at the moment we have `k8s-ag-ingress-ws.mis.li-80-rr`

Once this change goes in, and we keep the dash, this will become `-ws.mis.li-80-rr`
Without this dash it looks better - `ws.mis.li-80-rr`

So I decided to let the customer put a trailing dash in the prefix, should they decide to add one. (My intuition is - the prefix feature won't be used much)